libc 0.2.25 does not build on s390x-unknown-linux-gnu

```
error[E0412]: cannot find type `__u64` in the crate root
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.25/src/unix/notbsd/linux/mod.rs:283:26
    |
283 |           pub codes_ptr: ::__u64,
    |                            ^^^^^ not found in the crate root
    | 
   ::: /cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.25/src/lib.rs
    |
266 | / cfg_if! {
267 | |     if #[cfg(windows)] {
268 | |         mod windows;
269 | |         pub use windows::*;
...   |
278 | |     }
279 | | }
    | |_- in this macro invocation
error[E0204]: the trait `Copy` may not be implemented for this type
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.25/src/macros.rs:44:14
    |
44  |           impl ::dox::Copy for $i {}
    |                ^^^^^^^^^^^
    | 
   ::: /cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.25/src/unix/notbsd/linux/mod.rs
    |
283 |           pub codes_ptr: ::__u64,
    |           ---------------------- this field does not implement `Copy`
    | 
   ::: /cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.25/src/lib.rs
    |
266 | / cfg_if! {
267 | |     if #[cfg(windows)] {
268 | |         mod windows;
269 | |         pub use windows::*;
...   |
278 | |     }
279 | | }
    | |_- in this macro invocation
error: aborting due to previous error
```

As can be seen here: https://travis-ci.org/CryZe/livesplit-core/jobs/251173532#L311

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,5 @@
 language: rust
+rust: stable
 sudo: required
 dist: trusty
 services:
@@ -22,109 +23,63 @@ env:
 matrix:
   include:
     # 1.0.0 compat
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+    - env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
       rust: 1.0.0
       script: cargo build
       install:
 
     # build documentation
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+    - env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
       rust: nightly
       script: sh ci/dox.sh
 
     # stable compat
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
-      rust: stable
-    - os: linux
-      env: TARGET=i686-unknown-linux-gnu
-      rust: stable
+    - env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+    - env: TARGET=i686-unknown-linux-gnu
     - os: osx
       env: TARGET=x86_64-apple-darwin NO_ADD=1
-      rust: stable
     - os: osx
       env: TARGET=i686-apple-darwin
-      rust: stable
-    - os: linux
-      env: TARGET=arm-linux-androideabi
-      rust: stable
-    - os: linux
-      env: TARGET=aarch64-linux-android
-      rust: stable
-    - os: linux
-      env: TARGET=i686-linux-android
-      rust: stable
-    # as of 2017/05/03 x86_64-linux-android are not on stable
-    - os: linux
-      env: TARGET=x86_64-linux-android
-      rust: beta
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-musl
-      rust: stable
-    - os: linux
-      env: TARGET=i686-unknown-linux-musl
-      rust: stable
-    - os: linux
-      env: TARGET=arm-unknown-linux-gnueabihf
-      rust: stable
-    - os: linux
-      env: TARGET=aarch64-unknown-linux-gnu
-      rust: stable
+    - env: TARGET=arm-linux-androideabi
+    - env: TARGET=aarch64-linux-android
+    - env: TARGET=i686-linux-android
+    - env: TARGET=x86_64-linux-android
+    - env: TARGET=x86_64-unknown-linux-musl
+    - env: TARGET=i686-unknown-linux-musl
+    - env: TARGET=arm-unknown-linux-gnueabihf
+    - env: TARGET=aarch64-unknown-linux-gnu
     - os: osx
       osx_image: xcode8.2
       env: TARGET=i386-apple-ios
-      rust: stable
     - os: osx
       osx_image: xcode8.2
       env: TARGET=x86_64-apple-ios
-      rust: stable
-    - os: linux
-      env: TARGET=x86_64-rumprun-netbsd
-      rust: stable
-    - os: linux
-      env: TARGET=powerpc-unknown-linux-gnu
-      rust: stable
-    - os: linux
-      env: TARGET=powerpc64-unknown-linux-gnu
-      rust: beta
-    - os: linux
-      env: TARGET=mips-unknown-linux-musl
-      rust: stable
-    - os: linux
-      env: TARGET=mipsel-unknown-linux-musl
-      rust: stable
-    - os: linux
-      env: TARGET=mips64-unknown-linux-gnuabi64
-      rust: beta
-    - os: linux
-      env: TARGET=mips-unknown-linux-gnu
-      rust: beta
+    - env: TARGET=x86_64-rumprun-netbsd
+    - env: TARGET=powerpc-unknown-linux-gnu
+    - env: TARGET=powerpc64-unknown-linux-gnu
+    - env: TARGET=mips-unknown-linux-musl
+    - env: TARGET=mipsel-unknown-linux-musl
+    - env: TARGET=mips64-unknown-linux-gnuabi64
+    - env: TARGET=mips-unknown-linux-gnu
+    - env: TARGET=s390x-unknown-linux-gnu
 
     # beta
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+    - env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
       rust: beta
     - os: osx
       env: TARGET=x86_64-apple-darwin NO_ADD=1
       rust: beta
 
     # nightly
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
+    - env: TARGET=x86_64-unknown-linux-gnu NO_ADD=1
       rust: nightly
     - os: osx
       env: TARGET=x86_64-apple-darwin NO_ADD=1
       rust: nightly
 
     # QEMU based targets that compile in an emulator
-    - os: linux
-      env: TARGET=x86_64-unknown-freebsd
-      rust: stable
-    - os: linux
-      env: TARGET=x86_64-unknown-openbsd QEMU=openbsd.qcow2
-      rust: stable
+    - env: TARGET=x86_64-unknown-freebsd
+    - env: TARGET=x86_64-unknown-openbsd QEMU=openbsd.qcow2
       script: sh ci/run-docker.sh $TARGET
       install:
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 
 name = "libc"
-version = "0.2.25"
+version = "0.2.26"
 authors = ["The Rust Project Developers"]
 license = "MIT/Apache-2.0"
 readme = "README.md"

--- a/ci/docker/s390x-unknown-linux-gnu/Dockerfile
+++ b/ci/docker/s390x-unknown-linux-gnu/Dockerfile
@@ -1,0 +1,9 @@
+FROM ubuntu:17.04
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        gcc libc6-dev qemu-user ca-certificates \
+        gcc-s390x-linux-gnu libc6-dev-s390x-cross
+
+ENV CARGO_TARGET_S390X_UNKNOWN_LINUX_GNU_LINKER=s390x-linux-gnu-gcc \
+    CC_s390x_unknown_linux_gnu=s390x-linux-gnu-gcc \
+    PATH=$PATH:/rust/bin

--- a/ci/run.sh
+++ b/ci/run.sh
@@ -168,6 +168,11 @@ case "$TARGET" in
     qemu-aarch64 -L /usr/aarch64-linux-gnu/ $CARGO_TARGET_DIR/$TARGET/debug/libc-test
     ;;
 
+  s390x-unknown-linux-gnu)
+    # TODO: in theory we should execute this, but qemu segfaults immediately :(
+    # qemu-s390x -L /usr/s390x-linux-gnu/ $CARGO_TARGET_DIR/$TARGET/debug/libc-test
+    ;;
+
   *-rumprun-netbsd)
     rumprun-bake hw_virtio /tmp/libc-test.img $CARGO_TARGET_DIR/$TARGET/debug/libc-test
     qemu-system-x86_64 -nographic -vga none -m 64 \

--- a/src/unix/notbsd/linux/s390x.rs
+++ b/src/unix/notbsd/linux/s390x.rs
@@ -16,6 +16,7 @@ pub type greg_t = u64;
 pub type clock_t = i64;
 pub type __fsword_t = ::c_long;
 pub type __priority_which_t = ::c_uint;
+pub type __u64 = u64;
 
 s! {
     pub struct aiocb {
@@ -777,7 +778,6 @@ pub const CMSPAR: ::tcflag_t = 0o10000000000;
 pub const VSWTC: usize = 7;
 pub const OLCUC:  ::tcflag_t = 0o000002;
 pub const NLDLY:  ::tcflag_t = 0o000400;
-pub const NL1: ::tcflag_t  = 0x00000100;
 pub const CRDLY:  ::tcflag_t = 0o003000;
 pub const CR1: ::tcflag_t  = 0x00000200;
 pub const CR2: ::tcflag_t  = 0x00000400;

